src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -476,6 +476,19 @@ private void AdvanceReader(BufferSegment consumedSegment, int consumedIndex, Buf
             TrySchedule(_writerScheduler, completionData);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private long GetLength(BufferSegment startSegment, int startIndex, BufferSegment endSegment, int endIndex)
+        {
```

Comment:
Are the arguments validated before calling this method? This change remove a lot of validation that used to happen [in the ROS ctor](https://github.com/dotnet/corefx/blob/07d1bb7365ef9a4d44a25cd54e0db60f12acd6d2/src/System.Memory/src/System/Buffers/ReadOnlySequence.cs#L80-L86)

Reply:
That’s a good thing 👍🏿😬